themes/bootstrap5/js/record.js

```diff
@@ -487,14 +487,16 @@ function recordDocReady() {
   document.querySelectorAll('.record-tabs .nav-tabs a')
     .forEach((tab) => tab.addEventListener('click', (event) => {
       const li = tab.parentNode;
-      // Do nothing if the tab is already active:
-      if (tab.classList.contains('active')) {
-        event.preventDefault();
-        return;
-      }
       const tabId = li.dataset.tab;
       const top = tab.closest('.record-tabs');
       if (!top) return;
+      const targetPane = top.querySelector('.tab-pane.' + tabId + '-tab');
+      // Only trigger show on already active tabs to set up all attributes required for keyboard controls:
+      if (tab.classList.contains('active') && targetPane && targetPane.classList.contains('active')) {
+        event.preventDefault();
+        $(tab).tab('show');
+        return;
+      }
```

Comment:
I haven't tested this yet (and probably won't have time before next week due to the Thanksgiving holiday here), but I wonder if this would be a slightly safer refactor that achieves the same effect:

```suggestion
      // If the tab is already active, we need to return early:
      if (tab.classList.contains('active')) {
        event.preventDefault();
        // Trigger show on the already-active tab to set up all attributes required for keyboard controls:
        const targetPane = top.querySelector('.tab-pane.' + tabId + '-tab');
        if (targetPane && targetPane.classList.contains('active')) {
          $(tab).tab('show');
        }
        return;
      }
```

Unless I'm mistaken, this prevents an unlikely but theoretically possible edge case where the tab is active but the tab pane is missing, so the active tab triggers extra processing inappropriately.

Reply:
> an unlikely but theoretically possible edge case where the tab is active but the tab pane is missing

That is actually the scenario that happens when using the keyboard to select a tab. The tab is selected using the arrow keys and this logic is not triggered, because it was not selected by clicking on it. I added the `targetPane && targetPane.classList.contains('active')` part, so that the logic that is used to load the tab is not skipped in that case, when using "enter" to trigger the click listener.

Reply:
Ahh, sorry for the misunderstanding. I'll withdraw this suggestion. :-)